packages/compass-data-modeling/src/store/diagram.ts

```diff
@@ -227,22 +270,64 @@ export function renameDiagram(
   };
 }
 
-// TODO
-function _applyEdit(model: any, edit: any) {
-  if (edit && 'type' in edit) {
-    if (edit.type === 'SetModel') {
-      return edit.model;
+function _applyEdit(edit: Edit, model?: StaticModel): StaticModel {
+  if (edit.type === 'SetModel') {
+    return edit.model;
+  }
+  if (!model) {
+    throw new Error('Editing a model that has not been initialized');
+  }
+  switch (edit.type) {
+    case 'AddRelationship': {
+      return {
+        ...model,
+        relationships: [...model.relationships, edit.relationship],
+      };
+    }
+    case 'RemoveRelationship': {
+      return {
+        ...model,
+        relationships: model.relationships.filter(
+          (relationship) => relationship.id !== edit.relationshipId
+        ),
+      };
+    }
+    default: {
+      return model;
     }
   }
-  return model;
 }
 
-export function getCurrentModel(diagram: MongoDBDataModelDescription): unknown {
-  let model;
-  for (const edit of diagram.edits) {
-    model = _applyEdit(model, edit);
+export function getCurrentModel(
+  description: MongoDBDataModelDescription
+): StaticModel {
+  // Get the last 'SetModel' edit.
+  const reversedSetModelEditIndex = description.edits
+    .slice()
+    .reverse()
+    .findIndex((edit) => edit.type === 'SetModel');
```

Comment:
You can just [findLastIndex](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/findLastIndex) here

```suggestion
    .findLastIndex((edit) => edit.type === 'SetModel');
```

Reply:
I wanted to but looks like we first need to update ts config as this is ES2023.

Reply:
let me do that actually, hopefully I won't run into problems